### ISSUE TITLE
Login As not working after roles clean up

### ISSUE THREAD

**Issue by jgambarios** [dotCMS/core#15237]
After a clean up of roles the `Login As` functionality is not working as one of the removed roles was the `Administrator` role, that role should not be a required role and we can not assume that will always exist on the system.

```
java.lang.NullPointerException: null
    at com.dotcms.util.pagination.UserPaginator.getItems(UserPaginator.java:72) ~[dotcms_5.0.1_608da24.jar:?]
    at com.dotcms.util.pagination.UserPaginator.getItems(UserPaginator.java:56) ~[dotcms_5.0.1_608da24.jar:?]
    at com.dotcms.util.pagination.PaginatorOrdered.getItems(PaginatorOrdered.java:26) ~[dotcms_5.0.1_608da24.jar:?]
    at com.dotcms.util.PaginationUtil.getPage(PaginationUtil.java:128) ~[dotcms_5.0.1_608da24.jar:?]
    at com.dotcms.util.PaginationUtil.getPage(PaginationUtil.java:90) ~[dotcms_5.0.1_608da24.jar:?]
    at com.dotcms.rest.api.v2.user.UserResource.loginAsData(UserResource.java:65) ~[dotcms_5.0.1_608da24.jar:?]
    at sun.reflect.GeneratedMethodAccessor2130.invoke(Unknown Source) ~[?:?]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.internal.Errors.process(Errors.java:315) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.internal.Errors.process(Errors.java:297) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.internal.Errors.process(Errors.java:267) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) ~[dot.jersey-common-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) ~[dot.jersey-server-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471) ~[dot.jersey-container-servlet-core-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:425) ~[dot.jersey-container-servlet-core-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:383) ~[dot.jersey-container-servlet-core-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:336) ~[dot.jersey-container-servlet-core-2.22.1_4.jar:?]
    at com.dotcms.repackage.org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:223) ~[dot.jersey-container-servlet-core-2.22.1_4.jar:?]
    at com.dotcms.rest.servlet.ReloadableServletContainer.service(ReloadableServletContainer.java:98) ~[dotcms_5.0.1_608da24.jar:?]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
    at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:199) ~[dotcms_5.0.1_608da24.jar:?]
```

## Expected Behavior

The `Administrator` role should not be a required role.

## Current Behavior

If the `Administrator` role is removed the `Login As` does not work.
https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/util/pagination/UserPaginator.java#L72

**Comment by wezell** [comment]
The correct role should be `CMS Administrator`

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/15242

**Comment by bryanboza** [comment]
After the fix now we don't have the NPE, but we are unable to use the login-as feature. 

The system is requesting for a password and you are unable to login-as
https://content.screencast.com/users/Bryan_Boza/folders/Jing/media/cd659f9e-d840-4264-9654-503902dd8e3a/00000335.png

To test you need to go directly to the DB and delete the `Administrator` role using this query 
`delete from cms_role where role_name='Administrator';`

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/15302

**Comment by jdotcms** [comment]
<img width="930" alt="screen shot 2018-09-18 at 3 42 00 pm" src="https://user-images.githubusercontent.com/19578350/45719016-468f9380-bb5c-11e8-9f37-705353573e74.png">
<img width="1357" alt="screen shot 2018-09-18 at 3 39 37 pm" src="https://user-images.githubusercontent.com/19578350/45719017-47282a00-bb5c-11e8-936f-61d1256f66a4.png">
<img width="1355" alt="screen shot 2018-09-18 at 3 38 13 pm" src="https://user-images.githubusercontent.com/19578350/45719018-47282a00-bb5c-11e8-83eb-d0380bc4beba.png">
<img width="1347" alt="screen shot 2018-09-18 at 3 37 54 pm" src="https://user-images.githubusercontent.com/19578350/45719019-47282a00-bb5c-11e8-9d48-ab216af535db.png">

**Comment by jdotcms** [comment]
As you can see I have removed the Admin role by hand in the db. After that I have be able to get login as with diff users.

**Comment by bryanboza** [comment]
Now we have a NPE in the log after delete the Administrator role and try to use the login-as

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/15313

**Comment by bryanboza** [comment]
Fixed, tested on release-5.0.2 and now it works as should

**Comment by bryanboza** [comment]
Fixed, tested on master and now it is working as expected
